### ISSUE TITLE
openWindow is calling the URL twice in rapid succession when using TestCafe v1.10.1

### ISSUE THREAD

**Issue by iamle0pard** [DevExpress/testcafe-hammerhead#2544]
### What is your Test Scenario?
Open a window using the 'openWindow' command.  Expect that this URL is loaded only once.  Our system will block the request if the user tries to load the specified URL more than once every 30 seconds.
 
### What is the Current behavior?
After upgrading to TestCafe 1.10.1 from 1.9.4 we noticed that our tests started failing because the 'openWindow' call is making duplicate calls to the URL we pass it.  In our software this causes the UI to redirect to a "you must wait 30 seconds before visiting this page" error, which told us that the openWindow is making multiple requests.
 
### What is the Expected behavior?
We expect the 'openWindow' call to load the URL passed into it only a single time, just like it did in 1.9.4.
 
### What is your web application and your TestCafe test code?
Start Fiddler locally before running this test.
When you monitor fiddler, you will see that both "openWindow" urls are hit TWICE each.
**Please note:**
If you remove the first openWindow (to our own endpoint) you will see that the remaining openWindow URL is only hit ONCE.  I'm not sure if we're doing some redirect that is causing TestCafe to reload it and any subsequent calls, but this didn't happen with 1.9.4 but does now happen with 1.10.1

### Sample Code:
**sample.e2e.ts:**
```TypeScript
let childWindow;
let anotherWindow;

fixture('Sample test to reproduce issues...')
    .page('https://www.google.com')
    .beforeEach(async (t) => {
        childWindow = await t.openWindow(
            'https://bac-dev.ehosts.net/netagent/client/unified/desktop/naclient.aspx?LOGINNAME=TestCafe%20Dev%20Customer&PASSWORD=&QUEUE=109&SUBJECT=I%20am%20an%20E2E%20test%20customer%2C%20please%20help%20me!&EMAIL=e2e%40address.com&ENABLE_SEND_TRANSCRIPT=0&SENDCHATTRANSCRIPT=0&REFERER=https%3A%2F%2Fbac-dev.ehosts.net%2Fnetagent%2Flaunch.html&ROUTEIDENT=e2e&PROACTIVEID=&SALESVALUE=&LEADTYPE=&ROUTETOAGENT=0&LangSelection=11&PUSHPAGELOCATION=0&CHATFRAMESIZE=275&UNIFIEDCLIENTTEMP=desktop&PORTALID=EE7B3E9D-D25C-44B6-9AA9-3C142583DA3F&QUESTIONNAIREID=BB152583-8A6D-4B00-BB40-992CB87E0F68&WEBCOLLABKEY=&COBROWSE_ENABLED=0&FONT%5fCHOICE=1&defaultStyleId=EE7B3E9D-D25C-44B6-9AA9-3C142583DA3F&submitBtn=Start+Chat'
        );

        anotherWindow = await t.openWindow(
            'http://www.west.net/~jay/cheat.htm'
        )
    });

test('Test that will reproduce the issue.', async (t) => {
    console.log('Did we reproduce the issue?');
});
```
**runner.ts:**
```TypeScript
import createTestCafe from 'testcafe';

let testcafe: TestCafe;

const browsers = ['chrome:headless'];

async function runTestForBrowser(browser: string) {
    try {
        // Start running E2E tests
        testcafe = await createTestCafe('localhost');
        const runner = testcafe.createRunner();

        const testSource = ['./sample.e2e.ts'];

        // Log the current bacUrl to assist with debugging failed tests
        // console.log(`process.env.bacUrl: ${process.env.bacUrl}`);

        const failedCount = await runner
            .src(testSource)
            .browsers(browser)
            .reporter([
                'spec', // Output to STDOUT (console)
                {
                    name: 'xunit',
                    output: 'test/reports/bacUIe2e.xml',
                },
                {
                    name: 'html',
                    output: 'test/reports/bacUIe2e.html',
                },
            ])
            .screenshots({
                path: 'test/reports/screenshots/',
                takeOnFails: true,
                pathPattern: '${DATE}_${TIME}/test-${TEST_INDEX}/${USERAGENT}/${FILE_INDEX}.png',
            })
            .run({
                debugMode: false,
                debugOnFail: true,
                stopOnFirstFail: false,
                skipJsErrors: true,
                skipUncaughtErrors: true,
                selectorTimeout: 30000,
                assertionTimeout: 10000,
                pageLoadTimeout: 1000,
                speed: 1,
                disablePageCaching: false,
            })
            .catch((error) => {
                console.log(error);
                throw new Error(error);
            });

        if (failedCount > 0) {
            console.log('Tests failed: ' + failedCount);
            if (testcafe) {
                testcafe.close();
            }
            throw new Error(`TestCafe failed to run. Failed tests: ${failedCount}`);
        }

        return;
    } catch (error) {
        if (testcafe) {
            testcafe.close();
        }
        throw new Error(`TestCafe failed to run. ${error}`);
    }
}

const runAllBrowsers = async () => {
    for (const browser of browsers) {
        await runTestForBrowser(browser);
    }
    process.exit();
};

runAllBrowsers();

```

### Steps to Reproduce:
1. Launch a browser using the '.page' call
2. Launch a child browser using the 'openWindow' call
3. Notice that the 'openWindow' call made multiple requests to the url you passed in (you can see this using Fiddler to capture the requests while the test is running
 
### Your Environment details:
 
* testcafe version:                   1.10.1
* node.js version:                    10.15.0
* command-line arguments:  
* browser name and version: Chrome: Version 87.0.4280.88 (Official Build) (x86_64), Firefox: 83.0
* platform and version:          Windows 10, MacOS Catalina Version 10.15.7

**Comment by iamle0pard** [comment]
Thanks a lot for the follow up on this!

Do you guys happen to have an ETA or priority on fixing this issue?

It is currently blocking an entire suite of tests that we rely on (causing us to disable them and run the tests manually in the meantime).  We would just like to have some idea of when we might expect to see this fixed.

**Comment by SergeyShurygin** [comment]
We found the problematic commit ([https://github.com/DevExpress/testcafe-hammerhead/commit/7e9b280b672e2059b57f8723362eeb069ba90cc8](https://github.com/DevExpress/testcafe-hammerhead/commit/7e9b280b672e2059b57f8723362eeb069ba90cc8)). However, for now, I can't give you any estimates as to when the problem will be fixed.

**Comment by iamle0pard** [comment]
Is this fix available for us if we simply reference testcafe v1.10.1 in my package.json or do I need to do something specific to ensure that it is using the 'testcafe-hammerhead' version which contains this fix?

Sorry if this reactivates the issue, I didn't know how else to ask the question.

**Comment by Dmitry-Ostashev** [comment]
The fix for this issue was included in testcafe-hammerhead v19.2.0 and higher. You only need to update TestCafe to the newest 1.11.0 version, which contains a reference to a fixed testcafe-hammerhead.
